atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/thrift/ThriftObjectSizeUtils.java

```diff
@@ -284,15 +287,47 @@ private static <T> long getBytesSize(T data, Function<T, ByteBuffer> toByteBuffe
         return byteBuffer.remaining();
     }
 
-    public static <T> long getCollectionSize(Collection<T> collection, Function<T, Long> sizeFunction) {
+    public static <T> long getCollectionSize(@Nullable Collection<T> collection, ToLongFunction<T> sizeFunction) {
         if (collection == null) {
             return getNullSize();
         }
 
-        long sum = 0;
-        for (T item : collection) {
-            sum += sizeFunction.apply(item);
+        // random access lists can be more efficiently accessed via List::get(int)
+        // as this avoids allocating iterator
+        if (isRandomAccessList(collection)) {
+            return getCollectionSize((List<T>) collection, sizeFunction);
+        }
+        return sumSizes(collection, sizeFunction);
+    }
+
+    private static <T> long getCollectionSize(@Nullable List<T> list, ToLongFunction<T> sizeFunction) {
```

Comment:
I added this overload so that we avoid a bit of indirection for the list cases

![image](https://github.com/palantir/atlasdb/assets/54594/9c822a2b-a2a5-4ffb-b163-0f7aa1641d6b)

Reply:
Do we think we need to still check if the collection is a list then? Just b/c I think for the majority of places this is called, we know if it's a List or not.

Reply:
updated to simplify